src/utils/BrowserDetection.ts

```diff
@@ -14,3 +14,47 @@ export const isIE11 = () =>
 
 // Edge 20+
 export const isEdge = () => canUseDOM && !isIE() && 'styleMedia' in window;
+
+export const getChromeVersion = () => {
+  if (canUseDOM) {
+    const match = window.navigator.userAgent.match(/Chrome\/([0-9]+)\./);
+    return match ? parseFloat(match[1]) : false;
+  }
+
+  return false;
+};
+
+export const getSafariVersion = () => {
+  if (canUseDOM) {
+    const match =
+      !!window.navigator.userAgent.match(/Safari\//) &&
+      window.navigator.userAgent.match(/Version\/([0-9]+)\./);
+    return match ? parseFloat(match[1]) : false;
```

Comment:
<img width="1000" alt="image" src="https://user-images.githubusercontent.com/1203827/190045639-1a47d6ab-9327-4a4d-b2ec-9f1ab95b9bcc.png">

I recommend using `window.navigator.userAgent.match(/Version\/([\d\.]+).*Safari/) `. The version information it gets is more accurate. But I'm not sure if it has compatibility issues.

Reply:
there is a similar regexp from [detect-browser](https://github.com/DamonOehlman/detect-browser), the difference is their version support `_`. I think the current regexp works fine.
> https://github.com/DamonOehlman/detect-browser/blob/master/src/index.ts#L166

Reply:
👌